BanyanONNXRuntime/src/onnxruntime.jl

```diff
@@ -0,0 +1,34 @@
+function load_inference(path; kwargs...)
+    onnx_loc = RemoteONNXSource(path)
+    onnx_loc.src_name == "Remote" || error("$path does not exist")
+    onnx = Future(datatype="Onnx", source=onnx_loc)
+
+    (inputs, output_names=nothing) -> begin
+
+        # Right now, we only support a single input and output
+        # TODO: Support multiple inputs and outputs
+        # TODO: Support specifying inputs as Tuple
+
+        if length(inputs.keys()) > 1
+            error("Multiple inputs not supported")
+        end
+        if length(output_names) > 1
+            error("Multiple outputs not supported")
+        end
+
+        res = Future()
+        input = Future(inputs[inputs.keys()[1]])
+
+        partitioned_with() do
+            pt(res, Blocked(;along=1))
+            pt(onnx, Replicated())
+            pt(input, Blocked(;along=1))
+        end
+
+        @partitioned res onnx input
+            res = onnx(input)
+        end
+
+        res
+    end
+end
```

Comment:
You might want to be assigning the data processed by the model a `Blocked(along=1)` PT instead of using the `Blocked(Future)` constructor. Also, you could have separate PFs (calling a common helper function) that rely on `"balanced": true` or it being `false` with default `false` so that you can optimize to eliminate any extra communication/work to figure out how to uniquely name the files.